dev_guide/builds.adoc

```diff
@@ -1184,6 +1190,80 @@ spec:
 ----
 ====
 
+[[build-hooks]]
+
+== Build Hooks
+Build hooks can be used as a part of the build flow to run tests inside a temporary container and validate your image before it is being pushed to the registry. The command is executed with the current working directory set to the image's `WORKDIR`.
```

Comment:
Although we're introducing this as a means to run tests, it is worth mentioning that this is not to the exclusion of other possibilities.

- [ ] Briefly mention that this feature can be used for other purposes as well. Keep in mind that:
  1. The command runs in the same image that would be pushed to a registry
  2. If the command makes any changes, those won't be persisted. This can be "good" or "bad", depending on what you're trying to do, and knowing about this helps a user determine if this is the right feature she's looking for. You can install test dependencies prior to running tests, and those won't affect the final image. You cannot use this feature to "sign" a binary, or anything else one would expect to persist. You cannot use this to remove files produced during the build, etc.